### ISSUE TITLE
ES6 module with no exports is not found with --js_module_root

### ISSUE THREAD

**Issue by alexeagle** [google/closure-compiler#2247]
Repro:

```
$ npm install google-closure-compiler@20161201.0.0
$ cat user.js
import 'rxjs/add_first';

$ cat stripme/rxjs/add_first.js
import { Observable } from './Observable';
import { first } from './operator_first';
Observable.prototype.first = (first);

$ cat stripme/rxjs/operator_first.js
export function first() {}

$ cat stripme/rxjs/Observable.js
export class Observable {}

$ java -jar node_modules/google-closure-compiler/compiler.jar user.js stripme/rxjs/*.js --entry_point=user.js --js_module_root=stripme
user.js:1: ERROR - required "module$rxjs$add_first" namespace never provided
import 'rxjs/add_first';
^^^^^^^^^^^^^^^^^^^^^^^^

1 error(s), 0 warning(s)
```

It could be worked around in these ways:

1) add a trivial export to stripme/rxjs/add_first:
```
$ cat >> stripme/rxjs/add_first.js
export var a;

$ java -jar node_modules/google-closure-compiler/compiler.jar user.js stripme/rxjs/*.js --entry_point=user.js --js_module_root=stripme
var module$rxjs$Observable={},Observable$$module$rxjs$Observable=function(){};module$rxjs$Observable.Observable=Observable$$module$rxjs$Observable;var module$rxjs$operator_first={};function first$$module$rxjs$operator_first(){}module$rxjs$operator_first.first=first$$module$rxjs$operator_first;var module$rxjs$add_first={};Observable$$module$rxjs$Observable.prototype.first=module$rxjs$operator_first.first;var a$$module$rxjs$add_first;module$rxjs$add_first.a=a$$module$rxjs$add_first;
```

2) move the rxjs directory out of stripme/ and remove the `--js_module_root` argument.

**Comment by ChadKillingsworth** [comment]
I'm guessing a `goog.provide` statement is only generated when `export` is seen. Which is incorrect for ES6 modules.

**Comment by gregmagolan** [comment]
Update on this.

I tested with the latest closure compiler release (20170910) and the build still fails although the failures are different from the failure with 20161201 as described above. Different failures occur depending on configuration (BROWSER or NODE module resolution and relative and ambiguous import path in user.js). I was able to get a successful build with 20170910 but only with a specific configuration: module resolution NODE, relative import path in `user.js` and the correct order of `--js` files in `closure.conf`.

Using the closure compiler dist I made a few days ago (HEAD on 20170919 + patches) the build is successful with module resolution NODE. It worked with both a relative and ambiguous import path in `user.js` and the order of `--js` files in closure.conf didn't matter.

Code is here: https://github.com/gregmagolan/closure-test-2247. Test results are described in the README.
